pkg/platform/kube/apigatewayres/lazy.go

```diff
@@ -71,13 +74,21 @@ func (lc *lazyClient) Get(ctx context.Context, namespace string, name string) (R
 	return nil, errors.New("Method not implemented")
 }
 
-func (lc *lazyClient) CreateOrUpdate(ctx context.Context, apiGateway *nuclioio.NuclioAPIGateway) (Resources, error) {
+func (lc *lazyClient) CreateOrUpdate(ctx context.Context, apiGateway nuclioio.NuclioAPIGateway) (Resources, error) {
```

Comment:
Why is this change needed?

Reply:
@TomerShor That's a good question. Because we should not allow changing apiGateway itself - after using this method we do CRD Update (so we actually update all the fields). So, it's good to be on safe side here